rwkv/sampling.py

```diff
@@ -18,9 +18,10 @@ def sample_probs(probs: np.ndarray, temperature: float = 1.0, top_p: float = 0.8
     if logit_bias is not None:
         logits = np.log(probs)
 
-        for token in logit_bias.keys():
-            logits[token] += logit_bias[token]
-
+        if ids := list(logit_bias.keys()):
+            logits[ids] += logit_bias[ids]
+        
+        logits -= logits.max(axis=-1, keepdims=True)
```

Comment:
As I understand, this will subtract max logit from the logit array, basically converting it to range `min - max .. 0`. Why this is needed?

Reply:
"Why is* this needed". Numerical stability. You don't want e^(x >> 1) showing up. This is standard practice.

Reply:
Do you mean it is for avoiding `Inf` values with high positive logit biases?